src/lib.rs

```diff
@@ -404,65 +513,74 @@ impl<'arg, A: Argument + 'arg, I: Iterator<Item = A>> Options<A, I> {
                 Err(Error::DoesNotRequireValue(opt))
             }
 
-            State::Positional(_) | State::End => Ok(None),
+            State::Positional(_) | State::End { .. } => Ok(None),
         }
     }
 
-    /// Returns `true` if the last call to [`Options::next`] encountered
-    /// a `--`. In that case, it will have returned `None`, but without
-    /// this method you wouldn't be able to tell whether that was due to
-    /// the `--` or simply the next argument being positional (or not
-    /// existing).
+    /// Retrieves the next *argument*. An *argument* is represented by
+    /// [`Arg`] and defined as either an option ([`Opt`]) or a
+    /// positional argument.
     ///
-    /// This flag is not reset by [`Options::arg`].
+    /// This method is implemented entirely using other public methods
+    /// of [`Options`].
```

Comment:
I don't think this is necessary, it's just an implementation detail that's not useful for users.